### PR TITLE
Creation api connection part employ with csrf token validation

### PR DESCRIPTION
**Pull request by Nambinina7**

### CODE REVIEW

File: config/api_platform/resources/employee_onnection.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+App\Entity\EmployeeConnection:
```
Comment (Mokolos) [inline config/api_platform/resources/employee_onnection.yaml:1]: ito ve afak atambatra amin le iray csrf token iny ian passen parametre query le employee_connection sady tsy ampy c le anarana fichier

File: src/Controller/AuthEmployerController.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private FormFactoryInterface $formFactory;
+
+    public function __construct(CsrfTokenManagerInterface $csrfTokenManager, UserProviderInterface $userProvider, FormFactoryInterface $formFactory)
+    {
+        $this->csrfTokenManager = $csrfTokenManager;
+        $this->userProvider = $userProvider;
+        $this->formFactory = $formFactory;
+    }
+
+    #[Route(path: '/login/employee', name: 'api_employee_check', methods: ['POST'])]
```
Comment (Mokolos) [inline src/Controller/AuthEmployerController.php:34]: ito tokon /api ian f tokon public access ftsn any am security

File: src/Controller/CsrfTokenEmployeeConnectionController.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class CsrfTokenEmployeeConnectionController extends AbstractController
+{
+    public function __construct(
+        private readonly CsrfTokenManagerInterface $csrfTokenManager
+    ) {
+    }
+
+    public function __invoke(): JsonResponse
+    {
+        $csrfToken = $this->csrfTokenManager->getToken('authenticate');
```
Comment (Mokolos) [inline src/Controller/CsrfTokenEmployeeConnectionController.php:18]: le controller iray ian ampesaina d passeo paramètre query n authenticate na contact_ ...

File: src/Entity/EmployeeConnection.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?php
```
Comment (Mokolos) [inline src/Entity/EmployeeConnection.php:1]: mety tsy ilaina intsony ito fichier ito

File: src/Repository/EmployeeConnectionRepository.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?php
```
Comment (Mokolos) [inline src/Repository/EmployeeConnectionRepository.php:1]: mety tsy ilaina intsony ito fichier ito

File: config/packages/security.yaml [hunk 6]
```diff
@@ -5,6 +5,10 @@ security:
     # https://symfony.com/doc/current/security.html#loading-the-user-the-user-provider
     providers:
         # used to reload user from session & other features (e.g. switch_user)
+        api_user_provider:
```
Comment (Mokolos) [inline config/packages/security.yaml:8]: inon n mahasamihafa an ito amin io ambany io

File: config/packages/security.yaml [hunk 7]
```diff
@@ -15,6 +19,15 @@ security:
                 property: username
     firewalls:
         #Configuration lexik/jwt-authentication-bundle
+        employee:
+            pattern: ^/api/employee/login_check
```
Comment (Mokolos) [inline config/packages/security.yaml:23]: sod d tokon atreo amin /employee ian leiz?

File: src/Controller/CsrfTokenController.php [hunk 8]
```diff
@@ -2,22 +2,28 @@
...
+
     ) {
     }
 
     public function __invoke(): JsonResponse
     {
-        $csrfToken = $this->csrfTokenManager->getToken('contact_item');
+        $request = $this->requestStack->getCurrentRequest();
+
+        $tokenId = $request->query->get('tokenId');
```
Comment (Mokolos) [inline src/Controller/CsrfTokenController.php:24]: ovay _csrf_token le tokenId

File: config/packages/security.yaml (hunk 6 above)
Comment (Nambinina7) [reply config/packages/security.yaml:8]: Ka i Tendry le niteny oe api_user_provider ny provider d atoko oe mamorna anakiray zany satria tsy am reo

**Pull request merged**